tests/runall.m

```diff
@@ -1,25 +1,36 @@
-%% Script for running all Rigbox tests
-% To be called for code checks and the like
-% TODO May become a function
-% TODO May add flags for levels of testing
-% TODO Method setup in dat_test may become global fixture
+function [exitStatus, failures] = runall(ignoreTagged)
+% RUNALL gathers and runs all tests in Rigbox (including in the alyx-matlab
+% and signals submodules), and returns TestResults for failing tests.
+% To be called for code checks and the like.
+% 
+% Inputs:
+%   `ignoreTagged`: if true (default), tests that are tagged as requiring a
+%       specific hardware implementation will be ignored
+% Outputs:
+%   `exitStatus`: a number indicating whether all tests passed (0) or
+%       some tests failed (1)
+%   `failures`: an array of the failed tests
+%
+% TODO Method setup in `dat_test` may become global fixture
 % TODO Deal with directory changes
-main_tests = testsuite;
 
-%% Gather signals tests
-root = getOr(dat.paths,'rigbox');
-signals_tests = testsuite(fullfile(root, 'signals', 'tests'));
-
-%% Gather alyx-matlab tests
-alyx_tests = testsuite(fullfile(root, 'alyx-matlab', 'tests'));
+if nargin < 1; ignoreTagged = true; end
+import matlab.unittest.selectors.HasTag
+%% Gather tests
+rigbox_tests = testsuite;
```

Comment:
why did you make this change? someone may call `runall` from outside of the `tests` folder

Reply:
Which change precisely?  What I'm seeing here shouldn't be affected by working directory.